File: 12. Rekursi/C. Komposisi Fungsi II/c_komposisi_fungsi_ii.py
def fx(A, B, K, x):
    return abs(A * x + B)  if K == 1 \
        else abs(A * fx(A, B, K - 1, x) + B)

A, B, K, x = map(int, input().split())
print(fx(A, B, K, x))

File: 12. Rekursi/C. Komposisi Fungsi II/c_komposisi_fungsi_ii.cpp
#include <bits/stdc++.h>
using namespace std;

int fx(int A, int B, int K, int x)
{
    if (K == 1)
    {
        return abs(A * x + B);
    }
    else
    {
        return abs(A * fx(A, B, K - 1, x) + B);
    }
}

int main()
{
    int A, B, K, x;
    cin >> A >> B >> K >> x;
    cout << fx(A, B, K, x) << endl;
    return 0;
}
